packages/generator-clay-component/app/templates/_package.json

```diff
@@ -0,0 +1,58 @@
+{
+	"name": "@clayui/<%= repoName %>",
+	"version": "3.0.0",
+	"description": "<%= componentName %> component",
+	"license": "BSD-3-Clause",
+	"repository": "https://github.com/liferay/clay/tree/master/packages/<%= repoName %>",
+	"engines": {
+		"node": ">=0.12.0",
+		"npm": ">=3.0.0"
+	},
+	"main": "lib/<%= componentName %>.js",
+	"esnext:main": "src/<%= componentName %>.js",
+	"jsnext:main": "src/<%= componentName %>.js",
+	"files": ["lib", "src"],
+	"scripts": {
+		"build": "babel src --root-mode upward --out-dir lib --extensions .ts,.tsx",
+		"build:types": "tsc --project ./tsconfig.declarations.json",
+		"prepublish": "npm run build && npm run build:types",
+		"start": "webpack-dev-server --mode development",
+		"test": "jest --config ../../jest.config.js"
+	},
+	"keywords": ["clay", "react"],
+	"dependencies": {
+		"classnames": "^2.2.6"
+	},
+	"devDependencies": {
+		"@babel/cli": "^7.0.0",
+		"@babel/core": "^7.0.1",
+		"@babel/preset-env": "^7.0.0",
+		"@babel/preset-react": "^7.0.0",
+		"@babel/preset-typescript": "^7.0.0",
+		"@types/classnames": "^2.2.7",
+		"@types/jest": "^23.3.5",
+		"@types/react-dom": "^16.8.0",
+		"@types/react-test-renderer": "^16.8.0",
+		"@types/react": "^16.8.0",
+		"awesome-typescript-loader": "^5.2.1",
+		"clay-css": "^2.8.3",
+		"css-loader": "^2.1.0",
+		"html-webpack-plugin": "^3.2.0",
+		"jest": "^24.1.0",
+		"react-dom": "^16.8.1",
+		"react-test-renderer": "^16.6.1",
+		"react": "^16.8.1",
+		"source-map-loader": "^0.2.4",
+		"style-loader": "^0.23.1",
+		"ts-jest": "^23.10.4",
+		"typescript": "^3.0.3",
+		"webpack-cli": "^3.1.2",
+		"webpack-dev-server": "^3.1.10",
+		"webpack": "^4.29.0"
+	},
```

Comment:
Pretty much all dev dependencies can and I would argue should be hoisted up to the repo root. The node modules resolution algorithm means that `require('foo')` will still be found when working inside a package directory. Rationale is that we ensure that we are using a standard set of dev dependencies across all packages without any divergent versions. I know that you want developers to be able to work within package directories, but they still need to clone the repo in order to do that, so declaring the dev dependencies at the top level doesn't make working locally any harder. Take a look at a typical monorepo project like Babel and you'll see that all the dev dependencies are [at the top](https://github.com/babel/babel/blob/master/package.json), with only a couple of exceptions.

If you're interested, I have a checker in that sample project that [checks to make sure that dev dependencies only get declared at the top](https://github.com/wincent/js/blob/851033a694bd4c75dec463c9974f83e5c9d69b91/scripts/check-dependencies.js#L205-L236). Other checks are for things like [missing deps, mismatched deps etc](https://github.com/wincent/js/blob/851033a694bd4c75dec463c9974f83e5c9d69b91/scripts/check-dependencies.js#L285-L288).

Reply:
I had tried that originally but then I ran into issues with resolutions when I was in a package itself. Something couldve been out of wack so I will try that again though.

Reply:
I'd say it's worth it. Babel at least has [some docs](https://babeljs.io/docs/en/config-files#monorepos) on set-up for monorepos, although that might not help with the resolution issues you were having.